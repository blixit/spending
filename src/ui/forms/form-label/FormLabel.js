import React from 'react';
import styled from 'styled-components';

const LabelUI = styled.label`
  display: flex;
  flex-direction: column;
  font-size: 0.8rem;
  font-weight: bold;
  text-align: ${({ arabic }) => arabic ? 'right' : 'left'};
`;

const FormLabel = props => <LabelUI {...props} />;

FormLabel.displayName = 'FormLabel';

export default FormLabel;
