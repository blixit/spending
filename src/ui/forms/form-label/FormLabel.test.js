import React from 'react';
import 'jest-styled-components';

import render from 'testing/render';

import FormLabel from './FormLabel';

describe('[forms/form-label] FormLabel', () => {
  it('renders properly', () => {
    const tree = render(
      <FormLabel arabic >Label</FormLabel>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('checks the arabic props', () => {
    const tree = render(
      <FormLabel arabic >Label</FormLabel>
    ).toJSON();
    expect(tree).toHaveStyleRule('text-align', 'right');
  });
});
