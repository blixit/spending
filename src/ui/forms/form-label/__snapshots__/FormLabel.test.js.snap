// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[forms/form-label] FormLabel renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.8rem;
  font-weight: bold;
  text-align: right;
}

<label
  className="c0"
>
  Label
</label>
`;
